appsync/aws-appsync-events/src/main/java/com/amplifyframework/aws/appsync/events/EventsChannel.kt

```diff
@@ -133,24 +163,40 @@ class EventsChannel internal constructor(
         subscriptionId: String,
         authorizer: AppSyncAuthorizer
     ): Boolean = coroutineScope {
+        // create a deferred holder for subscription response
         val deferredSubscriptionResponse = async { getSubscriptionResponse(webSocket, subscriptionId) }
 
         // Publish subscription to websocket
-        publishSubscription(webSocket, subscriptionId, authorizer)
+        val queued = webSocket.sendWithAuthorizer(
+            webSocketMessage = WebSocketMessage.Send.Subscription.Subscribe(id = subscriptionId, channel = name),
+            authorizer = authorizer
+        )
+        if (!queued) {
+            webSocket.disconnectReason?.toCloseException() ?: ConnectionClosedException()
+        }
+
+        try {
+            webSocket.sendWithAuthorizer(
+                webSocketMessage = WebSocketMessage.Send.Subscription.Subscribe(id = subscriptionId, channel = name),
+                authorizer = authorizer
+            )
+        } catch (sendException: Exception) {
+            throw sendException
```

Comment:
Why bother catching something only to rethrow it?

Reply:
check the updates when I push. I had duplicated code and changed the way things were doing. Accidentally left an inbetween state. Thanks for finding.